src/Helpers/BasicAuth.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Dumbo\Helpers;
+
+use Dumbo\Context;
+
+class BasicAuth
+{
+    /**
+     * Create a middleware that checks for Basic Auth credentials in the Authorization header
+     *
+     * @param array $options The options for the basic authentication middleware
+     * @return callable The middleware
+     */
+    public static function basicAuth(array $options): callable
+    {
+        if (!isset($options['username']) && !isset($options['password']) && !isset($options['verifyUser']) && empty($options['users'])) {
+            throw new \Exception('Basic auth middleware requires options for "username and password" or "verifyUser" or "users"');
+        }
+
+        $realm = $options['realm'] ?? 'Secure Area';
+
+        return function (Context $ctx, callable $next) use ($options, $realm) {
+            $authHeader = $ctx->req->header('Authorization');
+
+            if (!$authHeader) {
+                return $ctx->json(
+                    ['error' => 'Authorization header missing'], 
+                    401, 
+                    ['WWW-Authenticate' => "Basic realm=\"$realm\""]
+                );
+            }
+            
+            $parts = explode(' ', $authHeader);
```

Comment:
If we implement the `unauthorizedResponse`, this could be simpler:

```php
$parts = explode(' ', $authHeader, 2);
if (count($parts) !== 2 || strtolower($parts[0]) !== 'basic') {
  return self::unauthorizedResponse($ctx, $realm, 'Invalid Authorization header format');
}
```